### PR TITLE
Added `-AsJob` parameter to `RemoveAzVMExtension` cmdlet

### PR DESCRIPTION
**Pull request by pselugar**
## Description

* Added `-AsJob` optional parameter to `Remove-AzVMExtension` cmdlet.
* Added Test case which does the following:
   1) Create a default VM in Azure portal.
   2) Use `Set-AzVMExtension`.
   3) Use `Remove-AzVMExtension`.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by VeryEarly — APPROVED** [review]

**Comment by pselugar** [comment]
> please let me know once it was ready to be merged

Yes, it is ready to be merged.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 1]
```diff
@@ -3132,6 +3132,63 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+       Set-AzVMExtension -ResourceGroupName $rgname -VMName $vmname -Publisher $publisher -Settings $publicConfig -ExtensionType $extensionType -ExtensionName $extensionName -Loc $loc -TypeHandlerVersion "1.0";
+
+       # Get VM
+       $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+       #Check for VmHealth Extension after removal
+       Remove-AzVMExtension -ResourceGroupName $rgname -Name $extensionName -VMName $vmname -Force;
+       $vmr = Get-AzVM -Name $vmname -ResourceGroupName $rgname;
+
+       Assert-Null $vmr.VMHealth;
```
[the remark this answers is not in the document]
Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3180]: You can expect an error with the type of syntax on line 3589 of this file, 
https://github.com/Azure/azure-powershell/blob/866d0e8f97aa758268f896a9fe62e93a152a9981/src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1#L3589

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 2]
```diff
@@ -3132,6 +3132,63 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+
+       # Remove
+       Remove-AzVM -Name $vmname -ResourceGroupName $rgname -Force;
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3183]: Why do this? the cleanup of the RG will handle this.

File: src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs [hunk 3]
```diff
@@ -58,6 +58,11 @@ public class RemoveAzureVMExtensionCommand : VirtualMachineExtensionBaseCmdlet
         [ValidateNotNullOrEmpty]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
```
Comment (Sandido) [inline src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs:64]: based on other cmdlets that use this AsJob, do you need do anything else with this parameter in this code?

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs (hunk 3 above)
Comment (pselugar) [reply src/Compute/Compute/Extension/RemoveAzureVMExtensionCommand.cs:64]: No in my understanding nothing else is needed

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 (hunk 2 above)
Comment (pselugar) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3183]: Good catch will remove that

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1 [hunk 4]
```diff
@@ -3132,6 +3132,59 @@ function Test-VirtualMachineGetStatusWithHealhtExtension
...
+       $extensionType = "ApplicationHealthWindows";
+       $publisher = "Microsoft.ManagedServices";
+       Set-AzVMExtension -ResourceGroupName $rgname -VMName $vmname -Publisher $publisher -Settings $publicConfig -ExtensionType $extensionType -ExtensionName $extensionName -Loc $loc -TypeHandlerVersion "1.0";
+
+       # Get VM
+       $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+       #Check for VmHealth Extension after removal
+       Remove-AzVMExtension -ResourceGroupName $rgname -Name $extensionName -VMName $vmname -Force;
+       Assert-ThrowsContains {
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1:3178]: Assert-ThrowsContains, what string is being provided to be expected? Right now it is expecting to find an "" in the string which will always be correct.

**Review by Sandido — CHANGES_REQUESTED** [review]

**Comment by VeryEarly** [comment]
Hi @Sandido , please approve once you finished reviewing.

**Comment by pselugar** [comment]
> Hi @Sandido , please approve once you finished reviewing.

@VeryEarly  , Adam approved, it's ready to merge

**Pull request merged**